src/parser.rs

```diff
@@ -176,6 +185,36 @@ impl Parser {
 
     /// Parse an expression prefix
     pub fn parse_prefix(&mut self) -> Result<Expr, ParserError> {
+        // PostgreSQL allows any string literal to be preceded by a type name, indicating that the
+        // string literal represents a literal of that type. Some examples:
+        //
+        //      DATE '2020-05-20'
+        //      TIMESTAMP WITH TIME ZONE '2020-05-20 7:43:54'
+        //      BOOL 'true'
+        //
+        // The first two are standard SQL, while the latter is a PostgreSQL extension. Complicating
+        // matters is the fact that INTERVAL string literals may optionally be followed by special
+        // keywords, e.g.:
+        //
+        //      INTERVAL '7' DAY
+        //
+        // Note also that naively `SELECT date` looks like a syntax error because the `date` type
+        // name is not followed by a string literal, but in fact in PostgreSQL it is a valid
+        // expression that should parse as the column name "date".
+        return_ok_if_some!(self.maybe_parse(|parser| match parser.parse_data_type()? {
+            DataType::Interval => parser.parse_literal_interval(),
+            // Single-quoted strings are parsed as custom data types, however this not desirable
+            // when we are handling input like `"NOT 'a' NOT LIKE 'b'"` because this will produce a
+            // TypedString instead of a SingleQuotedString. Further, this leads to issues where the
+            // same input will yield a BinaryOperator instead of the correct UnaryOperator. Here we
```

Comment:
I can remove this, it's just explaining why our test would fail if we didn't explicitly handle `DataType::Custom` by returning an error (e.g. it will parse as a BinaryOperator instead of an UnaryOperator).

Reply:
Would it be better to change this entire comment to say something to the effect of we don't accept the type 'string' syntax for the custom data types? (Borrowing from your comment below.)

Reply:
> It will parse as a BinaryOperator instead of an UnaryOperator
I don't understand _what_ will parse as BinaryOperator.

> Would it be better to change this entire comment to say something to the effect of we don't accept the type 'string' syntax for the custom data types? (Borrowing from your comment below.)

I think both the description of what returning an `Err()` here does and the non-obvious example of `NOT 'x'  LIKE 'y'` should stay in the comment. I'm just trying to make sure I understand the examples and that they are clearly described.

Reply:
> I don't understand what will parse as BinaryOperator.

Here I mean the direct example given in the comment of `"NOT 'a' NOT LIKE 'b'"`.

> I think both the description of what returning an Err() here does and the non-obvious example of NOT 'x' LIKE 'y' should stay in the comment. I'm just trying to make sure I understand the examples and that they are clearly described.

Sure, makes sense. I may be approaching the limits of my (admittedly shallow) understanding of the library here so I may need to defer to you for how you'd like to document that.

Reply:
> Here I mean the direct example given in the comment of "NOT 'a' NOT LIKE 'b'".

Ah, sorry for being dense.